### PR TITLE
Add Prometheus resource usage metrics for middleware

### PR DESCRIPTION
**Pull request by sethidden**
Sending a GET to /metrics will now output Prometheus metrics implemented by https://github.com/siimon/prom-client/
https://prometheus.io/docs/instrumenting/writing_clientlibs/#standard-and-runtime-collectors

- [ ] Expose metrics on separate port
- [ ] Add env var to disable prometheus

### CODE REVIEW

File: packages/middleware/src/helpers/metrics.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+const prometheusRegister = new prometheusClient.Registry();
+prometheusRegister.setDefaultLabels({
+  app: "middleware",
+});
+prometheusClient.collectDefaultMetrics({ register: prometheusRegister });
+
+/**
+ * Express API handler that should be registered on the /metrics route
+ * so that Prometheus can periodically call the middleware's /metrics endpoint
+ * and scrape the metrics
```
Comment (bartoszherba) [inline packages/middleware/src/helpers/metrics.ts:13]: What kind of metrics? This endpoint will be open (no token or whatever), anyone can hit it. Is it safe to expose such data?

File: packages/middleware/src/helpers/metrics.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { RequestHandler } from "express";
+import prometheusClient from "prom-client";
+
+const prometheusRegister = new prometheusClient.Registry();
+prometheusRegister.setDefaultLabels({
+  app: "middleware",
+});
+prometheusClient.collectDefaultMetrics({ register: prometheusRegister });
```
Comment (michaelKurowski) [inline packages/middleware/src/helpers/metrics.ts:8]: I think default metrics are metrics that are largely already available in console (CPU/Memory) no?

Comment (michaelKurowski) [reply packages/middleware/src/helpers/metrics.ts:8]: What would be useful is to keep average response time per endpoint

Comment (michaelKurowski) [reply packages/middleware/src/helpers/metrics.ts:8]: GC and heap could be useful I suppose, but yeah average response time per endpoint would definitely be useful as well. The page views initiative will also rely on prometheus

**Pull request closed without merging**